### PR TITLE
Add recipe for Physlr 

### PR DESCRIPTION
**Pull request by aafshinfard**
Adding Physlr to bioconda

Physlr: construct a physical map using linked reads
GitHub: https://github.com/bcgsc/physlr
DOI: https://doi.org/10.3390/dna2020009

----

* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/physlr/build.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+cd src
+make install
+cd ..
```
Comment (sjackman) [inline recipes/physlr/build.sh:5]:
```suggestion
make -C src install
```

File: recipes/physlr/build.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+mkdir -p ${PREFIX}/bin/
```
Comment (sjackman) [inline recipes/physlr/build.sh:7]:
```suggestion
```
Redundant with the following lines.

File: recipes/physlr/build.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+mkdir -p ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin
+mkdir -p ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/physlr
```
Comment (sjackman) [inline recipes/physlr/build.sh:9]:
```suggestion
```
Delete and see below

File: recipes/physlr/build.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+mkdir -p ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src
+
+cp bin/* ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin/
+cp physlr/* ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/physlr/
```
Comment (sjackman) [inline recipes/physlr/build.sh:13]:
```suggestion
cp -a bin ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/
cp -a physlr ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/
```

File: recipes/physlr/build.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+cp src/physlr-indexlr ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src/physlr-indexlr
+cp src/physlr-makebf ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src/physlr-makebf
+cp src/physlr-filter-bxmx ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src/physlr-indexlr
+cp src/physlr-filter-barcodes ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src/physlr-barcodes
+# cp src/physlr-molecules ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src/physlr-molecules
+cp src/physlr-overlap ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src/physlr-overlap
+cp src/physlr-split-minimizers ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src/physlr-split-minimizers
```
Comment (sjackman) [inline recipes/physlr/build.sh:21]:
```suggestion
cp src/physlr-indexlr src/physlr-makebf src/physlr-filter-bxmx src/physlr-filter-barcodes src/physlr-overlap src/physlr-split-minimizers ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src/
```
There appears to be unintentional copy-and-paste errors renaming files, like for example renaming `physlr-filter-bxmx` to `physlr-indexlr`.

File: recipes/physlr/build.sh [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+echo "#!/bin/bash" > ${PREFIX}/bin/physlr
+echo "make -f $(command -v ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin/physlr-make) \$@" >> ${PREFIX}/bin/physlr
```
Comment (sjackman) [inline recipes/physlr/build.sh:24]:
```suggestion
printf '#!/bin/bash\nexec make -f $(command -v '"${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin/physlr-make"') "$@" >${PREFIX}/bin/physlr
```

File: recipes/physlr/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    - tqdm
+    - make
+    - pigz
+    
+test:
+  commands:
+    - physlr version #| grep -q "{{ version }}"
+    # - physlr help | grep -q "physlr physical-map"
+    # - physlr help | grep -q "physlr scaffold"
+    # - physlr test
```
Comment (sjackman) [inline recipes/physlr/meta.yaml:42]: Either uncomment or delete these comments.

**Review by sjackman — CHANGES_REQUESTED** [review]

**Comment by aafshinfard** [comment]
Thanks for the corrections @sjackman; done; tests passing.

File: recipes/physlr/build.sh [hunk 8]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (sjackman) [inline recipes/physlr/build.sh:1]:
```suggestion
#!/bin/bash
set -eu
```
Fail on error or undefined variable.

**Comment by aafshinfard** [comment]
Local build with `bioconda-utils` worked alright.
Local build with `conda-build` was successful.
Local install with `conda install --use-local` was also successful, and I was able to run `physlr` afterward:
```
$ physlr version
physlr 1.0.2

**Comment by aafshinfard** [comment]
Awch, I found some issues I could not see as I had them already in my path. Will update shortly.

**Comment by aafshinfard** [comment]
@BiocondaBot please fetch artifacts

**Comment by aafshinfard** [comment]
@BiocondaBot please fetch artifacts

**Comment by aafshinfard** [comment]
@BiocondaBot please fetch artifacts

**Comment by aafshinfard** [comment]
@sjackman Made a new conda env and tested it on real human data and it worked alright

**Review by sjackman — APPROVED** [review]
LGTM

**Comment by aafshinfard** [comment]
@bioconda/core ready to be merged, please

**Comment by daler** [comment]
@BiocondaBot please merge

(thanks @aafshinfard @sjackman)

**Pull request merged**